File: serial.py
import time
import numpy as np
N =  10000000

def addVectors(v1:np.ndarray ,v2: np.ndarray):
    v3: np.ndarray = np.ones(N)
    for i in range(len(v1)):
        v3[i] = v1[i] + v2[i]
    return v3

def main():
    ones: np.ndarray = np.ones(N)
    negativeONes: np.ndarray = np.ones(N)*-1
    
    start = time.time()
    zeros: np.ndarray = addVectors(ones,negativeONes)
    end = time.time()
    print("Elapsed = %s" % (end - start))
    

    if(zeros.size != N):
        raise Exception("Summed Vector is Wrong Size")
    
    if not(np.array_equal(zeros,np.zeros(N))):
        raise Exception("Incorrect Summation for Zeros")

    print("Completed Summation")

main()

File: serial.c
#include <time.h>
#include <stdio.h>
#define N 10000000
// Set up
static int ones[N];
static int negativeOnes[N];
static int zeros[N];
static int sum[N];

void setUp(int* arr,int val){
    for (int i = 0; i < N; i++)
    {
        arr[i] = val;
    }
}

int main()
{
    setUp(ones, 1);
    setUp(negativeOnes, -1);
    setUp(zeros, 0);
    
    clock_t start, end;

    start = clock();
    for (int i = 0; i < N; i++)
    {
        sum[i] = ones[i] + negativeOnes[i];
    }
    end = clock();
    double duration = ((double)end - start) / CLOCKS_PER_SEC;
    printf("Time taken to execute in seconds : %f\n", duration);

    // Check Summation
    for (int i = 0; i < N; i++)
    {
        if (sum[i] != zeros[i])
        {
            printf("Not Summed Correctly");
        }
    }

    return 0;
}